pkg/model/model.go

```diff
@@ -149,19 +149,24 @@ type ScrapeContext struct {
 // CloudwatchData is an internal representation of a CloudWatch
 // metric with attached data points, metric and resource information.
 type CloudwatchData struct {
-	ID                      *string
-	MetricID                *string
-	Metric                  *string
-	Namespace               *string
-	Statistics              []string
-	Points                  []*Datapoint
-	GetMetricDataPoint      *float64
-	GetMetricDataTimestamps time.Time
-	NilToZero               *bool
-	AddCloudwatchTimestamp  *bool
-	Tags                    []Tag
-	Dimensions              []*Dimension
-	Period                  int64
+	ID                       string
```

Comment:
This is a very overloaded/poorly named field but I couldn't find a better name,

- In a discovery job this is the [resource arn](https://github.com/kgeckhart/yet-another-cloudwatch-exporter/blob/c4ddf3cc6c0ba5d08510dd992443b5ea25ce7a8b/pkg/job/discovery.go#L262-L262)
- In a custom namespace job this is the [custom namespace job name](https://github.com/kgeckhart/yet-another-cloudwatch-exporter/blob/c4ddf3cc6c0ba5d08510dd992443b5ea25ce7a8b/pkg/job/custom.go#L114-L114)
- In a static job this is the [resource name](https://github.com/kgeckhart/yet-another-cloudwatch-exporter/blob/c4ddf3cc6c0ba5d08510dd992443b5ea25ce7a8b/pkg/job/static.go#L29-L29)

`Resource` almost works except custom namespace jobs. It eventually is put on the metric in the `name` label but `name` doesn't feel any better

Reply:
`ResourceName`, and a nice comment on top clarifying the three possible meanings?